fix "incompatible pointer type" error

chirsz-ever opened the pull request:
GCC open -Wno-incompatible-pointer-types by default since C99, so I can't compile havoc with my GCC which use `-std=gnu23` as default standard.

We don't actually need the `noop` placeholder function, because the created proxies is of a lower version so libwayland will not require these callbacks in the lisenters to be non-empty.

ii8 commented:
I modified your PR to avoid this warning:

```
main.c:1365:1: warning: missing field 'axis_value120' initializer [-Wmissing-field-initializers]
 1365 | };
      | ^
```

Does it look good to you?

chirsz-ever commented:
In the future, if other callbacks are added to the listeners with the version increases, it will failed to compile again. `-Wmissing-field-initializers` does not warn about designated initializers[^1], so it is better to use designated initializers instead.

I updated the code and changed all lisenter initialization to use designated initializatiers. Please review it.

[^1]: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmissing-field-initializers

ii8 commented:
Merged, thank you.
